### PR TITLE
python3Packages.certbot: 1.12.0 -> 1.13.0

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
unbreak on hydra: https://hydra.nixos.org/job/nixpkgs/trunk/python38Packages.certbot.x86_64-linux
issue filed upstream: https://github.com/certbot/certbot/issues/8746

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/certbot/default.nix [hunk 1]
```diff
@@ -37,16 +37,19 @@ buildPythonPackage rec {
...
-    pytest
     pytestCheckHook
     pytest_xdist
   ];
 
-  pytestFlagsArray = [ "-o cache_dir=$(mktemp -d)" ];
+  pytestFlagsArray = [
+    "-o cache_dir=$(mktemp -d)"
+    # See https://github.com/certbot/certbot/issues/8746
+    "-W ignore::ResourceWarning"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/certbot/default.nix:51]: I need to remember that because this happens frequent with the latest pytest update. The alternative is to downgrade pytest which is a bit ugly when needing to override it multiple times.

@FRidh

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117734` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>simp_le: log was empty</li>
  </ul>
</details>
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>certbot (python38Packages.certbot)</li>
<li>certbot-full</li>
<li>home-assistant</li>
<li>python38Packages.acme</li>
<li>python38Packages.certbot-dns-cloudflare</li>
<li>python38Packages.certbot-dns-rfc2136</li>
<li>python38Packages.certbot-dns-route53</li>
<li>python38Packages.hass-nabucasa</li>
<li>python39Packages.acme</li>
<li>python39Packages.certbot</li>
<li>python39Packages.certbot-dns-cloudflare</li>
<li>python39Packages.certbot-dns-rfc2136</li>
<li>python39Packages.certbot-dns-route53</li>
<li>python39Packages.hass-nabucasa</li>
</ul>
</details>

**Review by dotlambda — APPROVED** [review]

File: pkgs/development/python-modules/certbot/default.nix [hunk 2]
```diff
@@ -37,16 +37,19 @@ buildPythonPackage rec {
...
   checkInputs = [
     dateutil
-    pytest
     pytestCheckHook
     pytest_xdist
   ];
 
-  pytestFlagsArray = [ "-o cache_dir=$(mktemp -d)" ];
+  pytestFlagsArray = [
+    "-o cache_dir=$(mktemp -d)"
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/certbot/default.nix:49]: @jonringer @FRidh

[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/certbot/default.nix:49]: So far, it seems we only need it for `certbot-*`, so let's only add it to `pytestCheckHook` if other packages' tests need it too.

[the remark this answers is not in the document]
Comment (FRidh) [reply pkgs/development/python-modules/certbot/default.nix:49]: pytest has a setup hook of itself as well, that cleans up the cache at the end of the build, regardless of whether it is invoked directly or through `pytestCheckHook`. We could let that setup hook always extend PYTEST_ADDOPTS to set `cache_dir`.

(note there is also the topic of why we have two separate hooks, but that's for in another place to discuss)

**Comment by risicle** [comment]
`simp_le` fails for me, linux x86_64:

```
builder for '/nix/store/b8lj5w9x3ww1r6lzvq7zacz9r5vvx77j-simp_le-client-0.17.0.drv' failed with exit code 1; last 10 log lines:
  installing
  Executing pipInstallPhase
  /build/simp_le-client-0.17.0/dist /build/simp_le-client-0.17.0
  Processing ./simp_le_client-0.17.0-py3-none-any.whl
  Requirement already satisfied: cryptography in /nix/store/g8x7nyna8kqx2jy9x9j05b66gpj8na0y-python3.8-cryptography-3.4.6/lib/python3.8/site-packages (from simp-le-client==0.17.0) (3.4.6)
  Requirement already satisfied: josepy in /nix/store/fv19b5y1nbirkfb7al82fvhj61baawhd-python3.8-josepy-1.7.0/lib/python3.8/site-packages (from simp-le-client==0.17.0) (1.7.0)
  Requirement already satisfied: pyOpenSSL in /nix/store/mfdfr6q8zzkbcrpqax8rdhpgrdkbs10a-python3.8-pyopenssl-20.0.1/lib/python3.8/site-packages (from simp-le-client==0.17.0) (20.0.1)
  Requirement already satisfied: acme>=1.0.0 in /nix/store/fdxl7q5x1ddh5fagqjyz5mjd2vw31slb-python3.8-acme-1.13.0/lib/python3.8/site-packages (from simp-le-client==0.17.0) (1.13.0)
  ERROR: Could not find a version that satisfies the requirement mock (from simp-le-client)
  ERROR: No matching distribution found for mock
```

**Comment by rmcgibbo** [comment]
@risicle: That looks unrelated to this PR to me.

**Comment by risicle** [comment]
It builds happily for me in `master` though.

**Comment by risicle** [comment]
Looks like `simp_le` was implicitly depending on `mock` being injected from here and now it's been removed here, it's failing. Probably need to add it explicitly to `simp_le`'s `buildInputs`, or patch it out and add it to its `checkInputs`.

File: pkgs/development/python-modules/acme/default.nix [hunk 3]
```diff
@@ -21,7 +21,7 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [
     cryptography pyasn1 pyopenssl pyRFC3339 pytz requests requests-toolbelt six
-    werkzeug mock ndg-httpsclient josepy
+    werkzeug ndg-httpsclient josepy
```
Comment (dotlambda) [inline pkgs/development/python-modules/acme/default.nix:24]: This change should be in a different commit.

**Review by dotlambda — CHANGES_REQUESTED** [review]
